src/esp/sensor/SensorFactory.cpp

```diff
@@ -9,9 +9,27 @@ sensor::SensorSuite SensorFactory::createSensors(
     scene::SceneNode& node,
     const sensor::SensorSetup& sensorSetup) {
   sensor::SensorSuite sensorSuite = sensor::SensorSuite();
-  for (const sensor::SensorSpec::ptr& spec : sensorSetup) {
+  for (const SensorSpec::ptr& spec : sensorSetup) {
     scene::SceneNode& sensorNode = node.createChild();
-    sensorSuite.add(sensor::CameraSensor::create(sensorNode, spec));
+    spec->sanityCheck();
```

Comment:
SensorFactory now calls the correct SanityCheck before creating Sensors

Reply:
It is not needed because when you create the sensor, it will be checked within the sensor's constructor.